commit e116fe2f7c9c79f9b5cf56932162fdeacb9c4229 (refs/changes/66/3366/2)
Author: Trent Mick <trentm@gmail.com>
Date:   2018-02-14T15:50:03-08:00 (1 year, 8 months ago)
    
    TRITON-145 improve SAPI's rsync-to to run a single rsync
    Reviewed by: Todd Whiteman <todd.whiteman@joyent.com>
    Approved by: Todd Whiteman <todd.whiteman@joyent.com>

diff --git a/tools/rsync-to b/tools/rsync-to
index 667a9bc..8c8c6fd 100755
--- a/tools/rsync-to
+++ b/tools/rsync-to
@@ -6,7 +6,7 @@
 #
 
 #
-# Copyright (c) 2015, Joyent, Inc.
+# Copyright (c) 2018, Joyent, Inc.
 #
 
 #
@@ -32,15 +32,18 @@ else
     ssh $NODE rm -rf /zones/$SAPI_ZONE/root/opt/smartdc/sapi/node_modules
 fi
 
-for f in lib node_modules sapi_manifests server.js smf test; do
-    rsync -av ${TOP}/$f $NODE:/zones/$SAPI_ZONE/root/opt/smartdc/sapi/ $extraOpts
-done
-#rsync -av ${TOP}/lib $NODE:/zones/$SAPI_ZONE/root/opt/smartdc/sapi/lib $extraOpts
-#rsync -av ${TOP}/node_modules $NODE:/zones/$SAPI_ZONE/root/opt/smartdc/sapi/node_modules $extraOpts
-#rsync -av ${TOP}/sapi_manifests $NODE:/zones/$SAPI_ZONE/root/opt/smartdc/sapi/sapi_manifests $extraOpts
-#rsync -av ${TOP}/server.js $NODE:/zones/$SAPI_ZONE/root/opt/smartdc/sapi/server.js $extraOpts
-#rsync -av ${TOP}/smf $NODE:/zones/$SAPI_ZONE/root/opt/smartdc/sapi/smf $extraOpts
-#rsync -av ${TOP}/test $NODE:/zones/$SAPI_ZONE/root/opt/smartdc/sapi/test $extraOpts
+rsync -av ${TOP}/ \
+    $NODE:/zones/$SAPI_ZONE/root/opt/smartdc/sapi/ \
+    $extraOpts \
+    --exclude .git/ \
+    --exclude /bin/ \
+    --exclude /boot/ \
+    --exclude /build/ \
+    --exclude /cmd/ \
+    --exclude /deps/ \
+    --exclude /docs/ \
+    --exclude /etc/ \
+    --exclude /tmp/
 
 state=$(ssh ${NODE} svcs -z ${SAPI_ZONE} -H -o state sapi)
 if [[ "$state" == "maintenance" ]]; then
